### DISCUSSION TITLE
[spaCy 3.0] NER training metrics zeroed-out

### DISCUSSION THREAD

**Discussion by fapbatista** [explosion/spaCy#7757] [Help: Coding & Implementations] [answered]
## Your Environment

- Operating System: Ubuntu 20.04
- Python Version Used: Python 3.8.8 on Anaconda env
- spaCy Version Used: 3.0.5
- Environment Information: Using GPU for training CUDA 10.2

I'm training a custom NER model (not using pipelines though, I'm using my own custom training loop as I used to do on spaCy 2.0). I partitioned my dataset in train/dev/test, I'm training on train set and evaluating on dev set each iteration. I keep getting zeroed metrics for ent_* key on Scorer object. Has anyone experienced any issues regarding NER metrics? 

Version 2.3.5 with a very similar approach seems to deliver metrics correctly.

Thanks on advance!

**Comment by svlandeg** [comment]
Hi! As a maintenance note - this type of question is well suited for the discussions forum, so I'll move it there. The original issue will be closed, but will link through to the open discussion.

**Comment by svlandeg** [comment, marked as answer]
I don't think I've seen the behaviour you describe before. Can you share some more details on how exactly you're running your experiments, preferably some example code and/or console output from the training log?

**Reply by fapbatista** [reply]
Hello @svlandeg, sure. This is what I'm using, I migrated the script I was using on spaCy 2.x basically. The changes are regarding GoldParse deprecated class in favor of the more Generic Example class. And also, the training loop regarding batching changes to use a Iterator compatible subclass. Other than this, it's just spaCy 2 NER example portrayed here.

[Edit] Put on a gist, it's readable now. Sorry.
https://gist.github.com/fapbatista/08d1cda34f87420a8442d9d55daa0de0

**Reply by fapbatista** [reply]
Adding more information: the training is apparently working though, after some epochs it starts to predict correctly entites in the fixed sample in the code. The metrics, F1, precision and recall, both general and entity-specific appear zeroed-out though. I'm digging through the code of the Scorer class to see if I can detect something in my code that may be causing it. Thanks in advance.

**Reply by fapbatista** [reply]
@svlandeg Ok, solved it. As customary I misunderstood some of the new concepts regarding Example class. Thanks for your attention and quick response to my issue. I was using the same function to generate training samples and testing examples. I was using the non-trained nlp object to generate the Doc input to each Example class, this caused the Example instances generated to have no x field to compare to the y field of the training instances, thus the zero score returned by the scorer. Thanks again!

**Reply by svlandeg** [reply]
Happy to hear you got it resolved, and thanks for reporting back! It seems I can't select your answer as correct (because it's a subthread?) but I'll go ahead and mark the whole thread as answer just so we know this is not an open question anymore ;-)
